Update CI config

Comandeer opened the pull request:
<!--
🚨 If you want to submit a PR for a security vulnerability, please contact us directly
at https://ckeditor.com/contact/ instead. 🚨
-->
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [ ] Unit tests
- [ ] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

N/A

## What changes did you make?

I've removed the Travis CI config and replaced it with the Circle CI one. The main change is the fact that now all runs are full runs (the diffing is disabled).

## Which issues does your PR resolve?

N/A

KarolDawidziuk commented on .circleci/config.yml:
```diff
@@ -0,0 +1,70 @@
+jobs:
+  test_chrome:
+    docker:
+      - image: cimg/node:10.24.0-browsers
```
```suggestion
      - image: cimg/node:10.24.1-browsers
```
In Travis.ci we used a newer version of Node. To be consistent let's update it.

KarolDawidziuk reviewed (changes requested):
Hi @Comandeer 

Tests are running correctly and CI configuration looks good ✅ 

There are some smaller things to improve.
* In the `[bender.ci.config](https://github.com/ckeditor/ckeditor4/blob/circle-ci/bender.ci.js#L7)` file we have an `isTravis` option which is no longer valid. 
Could you check if we can remove this option? Please remember to update the tests where this property was used.
* Looks like the `dev/travis/` directory can be removed.
* Travis badges in README files should be replaced by the [circle-ci one](https://app.travis-ci.com/ckeditor/ckeditor4).

Comandeer commented:
@KarolDawidziuk, thx for the review.

*   Updated the Docker image.
*   Removed badge and replaced it with CircleCI's one.
*   Removed Travis-oriented scripts.
*   Removed `config.isTravis` variable from Bender's config and reenabled tests.
*   Updated CKE4 build config to ignore `.circleci` dir instead of the `.travis.yml` file.

KarolDawidziuk reviewed (approved):
Great job @Comandeer 

LGTM! 🎉
